File: Baha/elevation/src/com/esprit/models/Seance.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esprit.models;

import java.sql.Date;

/**
 *
 * @author NADHIR
 */
public class Seance {

 
    
    private int idSeance;
    private String nomClasse;
    private String dateSeance;
    private String heure;
    private String matiere;
    private String nomEnseignant;
    
       public Seance(int idSeance, String nomClasse, String dateSeance, String heure, String matiere, String nomEnseignant) {
        this.idSeance = idSeance;
        this.nomClasse = nomClasse;
        this.dateSeance = dateSeance;
        this.heure = heure;
        this.matiere = matiere;
        this.nomEnseignant = nomEnseignant;
    }

    public Seance(String nomClasse, String dateSeance, String heure, String matiere, String nomEnseignant) {
        this.nomClasse = nomClasse;
        this.dateSeance = dateSeance;
        this.heure = heure;
        this.matiere = matiere;
        this.nomEnseignant = nomEnseignant;
    }

    public Seance(String dateSeance, String heure, String matiere, String nomEnseignant) {
        this.dateSeance = dateSeance;
              this.heure = heure;

        this.matiere = matiere;
        this.nomEnseignant = nomEnseignant;
    }

//  public Seance(String dateSeance, String heure, String matiere, String nomClasse) {
//        this.dateSeance = dateSeance;
//              this.heure = heure;
//
//        this.matiere = matiere;
//        this.nomClasse = nomClasse;
//    }
    public Seance(int idSeance, String nomClasse, String matiere, String nomEnseignant) {
        this.idSeance = idSeance;
        this.matiere = matiere;
        this.nomClasse = nomClasse;
        this.nomEnseignant = nomEnseignant;
    }


   

    public String getDateSeance() {
        return dateSeance;
    }

    public void setDateSeance(String dateSeance) {
        this.dateSeance = dateSeance;
    }

    public String getHeure() {
        return heure;
    }

    public void setHeure(String heure) {
        this.heure = heure;
    }

    

  
 

    public String getNomClasse() {
        return nomClasse;
    }

    public void setNomClasse(String nomClasse) {
        this.nomClasse = nomClasse;
    }

    public String getNomEnseignant() {
        return nomEnseignant;
    }

    public void setNomEnseignant(String nomEnseignant) {
        this.nomEnseignant = nomEnseignant;
    }

    public Seance( String dateSeance, String heure, String nomClasse) {
        this.nomClasse = nomClasse;
        this.dateSeance = dateSeance;
        this.heure = heure;
    }



    public Seance(int idSeance, String matiere, String nomClasse) {
        this.idSeance = idSeance;
        this.matiere = matiere;
        this.nomClasse = nomClasse;
    }

   


    public Seance(String nomEnseignant) {
        this.nomEnseignant = nomEnseignant;
    }


  
    public Seance(int idSeance) {
        this.idSeance = idSeance;
    }

    public Seance() {
    }

    public int getIdSeance() {
        return idSeance;
    }

    public void setIdSeance(int idSeance) {
        this.idSeance = idSeance;
    }


    
    public String getMatiere() {
        return matiere;
    }

    public void setMatiere(String matiere) {
        this.matiere = matiere;
    }

    @Override
    public String toString() {
        return "Seance{" + "idSeance=" + idSeance + ", nomClasse=" + nomClasse + ", dateSeance=" + dateSeance + ", heure=" + heure + ", matiere=" + matiere + ", nomEnseignant=" + nomEnseignant + '}';
    }


 
   
}

File: src/Entity/Seance.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Entity\Enseignant;
use Symfony\Component\Validator\Constraints as Assert;



/**
 * Seance
 *
 * @ORM\Table(name="seance", indexes={@ORM\Index(name="seance_ibfk_1", columns={"idEns"})})
 * @ORM\Entity
 */
class Seance
{
    /**
     * @var int
     *
     * @ORM\Column(name="idSeance", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idseance;

    /**
     * @var string
     *
     * @ORM\Column(name="nomClasse", type="string", length=20, nullable=false)
     *
     */
    private $nomclasse="";

    /**
     * @var string
     *
     * @ORM\Column(name="matiere", type="string", length=20, nullable=false)
     *
     */
    private $matiere="";

    /**
     * @var string
     *
     * @ORM\Column(name="heure", type="string", length=20, nullable=false)
     */
    private $heure="";

    /**
     * @var string
     *
     * @ORM\Column(name="dateSeance", type="string", length=20, nullable=false)
     */
    private $dateseance="";

    /**
     * @var \Enseignant
     * @Assert\NotBlank

     * @ORM\OneToOne(targetEntity="Enseignant")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idEns", referencedColumnName="idEns")
     * })
     */
    private $idens;

    /**
     * @return int
     */
    public function getIdseance(): int
    {
        return $this->idseance;
    }

    /**
     * @param int $idseance
     */
    public function setIdseance(int $idseance): void
    {
        $this->idseance = $idseance;
    }

    /**
     * @return string
     */
    public function getNomclasse(): string
    {
        return $this->nomclasse;
    }

    /**
     * @param string $nomclasse
     */
    public function setNomclasse(string $nomclasse): void
    {
        $this->nomclasse = $nomclasse;
    }

    /**
     * @return string
     */
    public function getMatiere(): string
    {
        return $this->matiere;
    }

    /**
     * @param string $matiere
     */
    public function setMatiere(string $matiere): void
    {
        $this->matiere = $matiere;
    }

    /**
     * @return string
     */
    public function getHeure(): string
    {
        return $this->heure;
    }

    /**
     * @param string $heure
     */
    public function setHeure(string $heure): void
    {
        $this->heure = $heure;
    }

    /**
     * @return string
     */
    public function getDateseance(): string
    {
        return $this->dateseance;
    }

    /**
     * @param string $dateseance
     */
    public function setDateseance(string $dateseance): void
    {
        $this->dateseance = $dateseance;
    }

    /**
     * @return \Enseignant
     */
    public function getIdens(): ?Enseignant
    {
        return $this->idens;
    }

    /**
     * @param \Enseignant $idens
     */
    public function setIdens(Enseignant $idens): self
    {
        $this->idens = $idens;
        return $this;
    }


}
